bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/presence/NAPresenceCameraHandler.java

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.netatmo.internal.presence;
+
+import static org.openhab.binding.netatmo.internal.ChannelTypeUtils.toOnOffType;
+
+import io.swagger.client.model.NAWelcomeCamera;
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.i18n.TimeZoneProvider;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.openhab.binding.netatmo.internal.camera.CameraHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import static org.openhab.binding.netatmo.internal.NetatmoBindingConstants.CHANNEL_CAMERA_FLOODLIGHT;
+import static org.openhab.binding.netatmo.internal.NetatmoBindingConstants.CHANNEL_CAMERA_FLOODLIGHT_AUTO_MODE;
+
+/**
+ * {@link NAPresenceCameraHandler} is the class used to handle Presence camera data
+ *
+ * @author Sven Strohschein
+ */
+@NonNullByDefault
+public class NAPresenceCameraHandler extends CameraHandler {
+
+    private static final String PING_URL_PATH = "/command/ping";
+    private static final String FLOODLIGHT_SET_URL_PATH = "/command/floodlight_set_config";
+
+    private final Logger logger = LoggerFactory.getLogger(NAPresenceCameraHandler.class);
+
+    private Optional<CameraAddress> cameraAddress = Optional.empty();
+    private State floodlightAutoModeState = UnDefType.UNDEF;
+
+    public NAPresenceCameraHandler(final Thing thing, final TimeZoneProvider timeZoneProvider) {
+        super(thing, timeZoneProvider);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String channelId = channelUID.getId();
+        switch (channelId) {
+            case CHANNEL_CAMERA_FLOODLIGHT:
+                if (OnOffType.ON.equals(command)) {
```

Comment:
== OnOffType.ON could rather ne used I think.

Reply:
I don't prefer and wouldn't recommend it, because this can get easily broken without a notice, for example when someone refactors the enum to a class with constants or when a new version of the swagger generator doesn't generate enums anymore. The usages aren't broken but at runtime it would not work anymore. An it's hard to find these usages because you can't search type-safe for "==" within big code bases.

But I changed it... ;-)

Done.